### DISCUSSION TITLE
Lesson 1: Simple Storage

### DISCUSSION THREAD

**Discussion by GruntRush82** [smartcontractkit/full-blockchain-solidity-course-py#430] [Q&A] [answered]
Hi, 
I am having an issue with the first contract simpleStorage. 
I am unable to add an uint to an array. The contract will compile but anytime that I try and view the array I get the following error:
![image](https://user-images.githubusercontent.com/94949398/143178016-d9922c32-e198-4dd3-ad7b-f1b454d9b645.png)
I get this error even when I copy the code directly from the sample contract.
I also created a test contract that simple initializes an array with units and it pulls the same error:

pragma solidity ^0.8.0;

contract Test {
    uint[2] public numberArray = [125.21,546.24];
}

I don't really know what this BigNumber error is, it must be something local because I the sample contract also causes this error.
Any thoughts?

**Comment by cromewar** [comment, marked as answer]
Hello.

125.21,546.24 are not valid numbers, as they are float and solidity does not support natively floating numbers, sending the numbers as string should do the work though.

Optionally yo can manage those numbers using web3 check [this](https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html?highlight=Bignumber#id14) as reference.

Best regards, Cromewar.

**Reply by GruntRush82** [reply]
Thanks for the tip. The issue was that I needed to call the variable with a parameter indicating the index of the array I was trying to pull.

**Reply by cromewar** [reply]
Did you solve it tough? @GruntRush82
